File: backend/app/models/blog.rb
class Blog < ApplicationRecord
  belongs_to :user
  has_one :notion, as: :notionable, dependent: :destroy
  validates :title, presence: true
  validates :summary, presence: true, length: { maximum: 500 }

  delegate :key, to: :notion, prefix: true, allow_nil: true
end

File: frontend/src/models/blog.ts
export default class Blog {
  constructor(
    private _title: string,
    private _summary: string,
    private _notionKey: string
  ) {}

  get title() {
    return this._title;
  }

  get summary() {
    return this._summary;
  }

  get notionKey() {
    return this._notionKey;
  }
}
